src/jimple-frontend/AST/jimple_ast.h

```diff
@@ -0,0 +1,193 @@
+/*******************************************************************\
+Module: Jimple AST Interface
+Author: Rafael Sá Menezes
+Date: September 2021
+Description: This interface will define every method that needs to
+  be implemented by every Jimple AST
+\*******************************************************************/
+
+#ifndef ESBMC_JIMPLE_AST_H
+#define ESBMC_JIMPLE_AST_H
+
+#include <util/message/default_message.h>
+#include <util/expr.h>
+#include <util/context.h>
+#include <util/std_types.h>
+#include <util/c_types.h>
+#include <nlohmann/json.hpp>
+
+// For json parsing
+using json = nlohmann::json;
+
+/**
+ * @brief Base interface for Jimple AST
+ */
+class jimple_ast
+{
+public:
+  /**
+   * @brief Prints the contents into the
+   * the stdout.
+   */
+  void dump() const
+  {
+    default_message msg;
+    msg.debug(this->to_string());
+  }
+
+  /**
+   * @brief Initializes the current instance by parsing
+   * a JSON
+   *
+   * @param json The json object relative to the structure
+   */
+  virtual void from_json(const json &) = 0;
+
+  /**
+   * @brief Converts the object into a string
+   *
+   * @return a human readable string of the object
+   */
+  virtual std::string to_string() const = 0;
+
+protected:
+  /**
+   * @brief creates a symbol with the default characteristics
+   *
+   * TODO: Right now, we are setting it as C mode. In time, we
```

Comment:
Explain what is needed to do it for Jimple.

Reply:
I've added a comment explaining